### DISCUSSION TITLE
Error when trying to import Pocketbase module

### DISCUSSION THREAD

**Discussion by Alextub** [pocketbase/pocketbase#5023] [Q&A] [answered]
Hi everyone,

First of all, I am very new to the developing world, so sorry in advance if I'm being unclear or if I am asking a really silly question.

I am trying to work on a project, using PocketBase as a back-end. You can find a sample of my project here : https://github.com/Alextub/test_pocketbase

I have added the following script tag in **index.html** :
```html
<script src="/js-sdk-master/dist/pocketbase.umd.js"></script>
``` 

And imported pocketabse in **main.js** : 
```javascript
import PocketBase from 'pocketbase';
``` 

But when launching it, I get the following error in the console: _TypeError: Failed to resolve module specifier "pocketbase". Relative references must start with either "/", "./", or "../"._

I guess I probably have forgotten something in the configuration or something but I can't find out how to solve this issue.

Thanks a lot for your help!

**Comment by ganigeorgiev** [comment]
It is not clear from the description, nor the above repo what actually is the intended behavior.

If you are trying to import the SDK in a browser, then as explained in [Browser (manually via script tag)](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#browser-manually-via-script-tag), you have 2 options:
1. With the UMD bundle (_notice that there is no import statements with this_):
```html
<script src="/path/to/dist/pocketbase.umd.js"></script>
<script type="text/javascript">
    const pb = new PocketBase("https://example.com")
    ...
</script>
```

2. With the ESM bundle (_recommended; notice the `type="module"` attribute and the es.mjs file_):
```html
<script type="module">
    import PocketBase from '/path/to/dist/pocketbase.es.mjs'

    const pb = new PocketBase("https://example.com")
    ...
</script>
```

If the above doesn't help, please elaborate more clearly what you are trying to do.

**Reply by ganigeorgiev** [reply]
If you choose the second approach and don't want to type every time the path to the esm bundle and just want `import PocketBase from "pocketbase"` then you can define an import map. More details and example about this you can find in https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap#bare_modules.

**Reply by Alextub** [reply]
Hello, thanks for your reply!

Yes, sorry if I was unclear but indeed, I am trying to import the SDK in a browser. To be more explicit, I am first trying to use the create API to upload data from my HTML form to my PocketBase DB.

I am currently trying to use the UMD bundle, and added the line `const pb = new PocketBase('http://127.0.0.1:8090');` into my js file **main.js**

Let me know if you need any further information!

**Reply by ganigeorgiev** [reply, marked as answer]
If you use the UMD bundle then doing something like `const pb = new PocketBase('http://127.0.0.1:8090');` should work.
In your initial post it doesn't work because you have an `import` statement and they work only with ES modules.

Or in other words, remove the `import` statement from here - https://github.com/Alextub/test_pocketbase/blob/trunk/main.js#L1C1-L1C37 (_and maybe also remove the `type="module"` from the `index.html` script tag since the imported js file is not a ES module_).

**Reply by Alextub** [reply]
It worked! I removed the `import` in **main.js** and the `type='module` and I no longer have this error.

Thanks a lot for your quick help 🙌
